tasks/test_e2e.yml

```diff
@@ -1,39 +1,66 @@
-    - name: "Extract URL from stf-connectors template"
-      shell: grep interconnect /home/stack/virt/stf-connectors* | sed 's;.*\(-service-telemetry.*\);default-prometheus-proxy\1;'
-      register: prom_url
-      failed_when: "prom_url.stdout_lines|length == 0"
+    - name: "Copy quicklab key"
+      shell: curl https://gitlab.cee.redhat.com/cee_ops/quicklab/raw/master/docs/quicklab.key --output ~/.ssh/quicklab.key; chmod 600 ~/.ssh/quicklab.key
+          
+    - name: "Extract STF FQDN"
+      shell: grep interconnect /home/stack/virt/stf-connectors* | sed '/\:*\./s/^[^.]*\ //'
+      register: stf_fqdn
     - debug:
-        var: prom_url.stdout_lines
+        var: stf_fqdn.stdout
+ 
+    - name: Get the default-prometheus-htpasswd secret
+      shell:
+        cmd: |
+          ssh -i ~/.ssh/quicklab.key   {{ stf_user }}@{{ stf_host }}  oc get secret default-prometheus-htpasswd -ojson | jq '.data.auth, .data.password' |  sed 's/"//g'
+      register: prom_secret
 
+  
     - debug:
-        msg: The URL is {{ prom_url.stdout_lines }}
+        var: prom_secret | string
 
+    - name: "Get the prom URL"
+      shell:
+        cmd: |
+          ssh -i ~/.ssh/quicklab.key  {{ stf_user }}@{{ stf_host }}  oc get route/default-prometheus-proxy -ojson | jq '.spec.host' | sed 's/"//g'
```

Comment:
This can be configured as a host in an inventory file. Currently, these are run on the group ``overcloud_nodes``

It makes sense to add a node which can access the ``oc`` too and kubeconfig. Each host can be in multiple groups.

the file will look something like this

```
# ... other groups here
# node with access to the OpenShift cluster
[stf]
<hostname> ansible_host="<ipaddress>" ansible_user="<user>" ansible_ssh_private_key_file="<path/to/private/key"
```

In this case it would look like this:
```
[stf]
my_stf_hostname ansible_host="$stf_host" ansible_user="$stf_user" ansible_ssh_private_key_file="/home/user/.ssh/quicklab.key"
```

Reply:
e2e tests (prometheus queries) run on undercloud

Reply:
In these tasks, because you're using ssh in the shell command, this is equivalent to running them on the stf_host.

Reply:
Hey,Emma ! Regarding inventory file. When we run functional tests it uses IR inventory file. How can I add STF host in the inventory ? Also if I understood correctly if we add it to inventory the name of the host will be hardcoded ,so if I will have different  cluster with different name it won't work.

Reply:
You can add multiple inventory files for ansible, so there wouldn't be a need to update the infrared one.

For changing hostnames, you would need a new inventory file for each deployment of STF, but that would be equivalent to using a different inventory file per infrared deployment as well. 
The test run shouldn't care about it, and just use the inventories that are pointed at it. We would ideally have  pre-described group names i.e. overcloud, undercloud, compute, controller (like in the infrared inv file) and stf (as proposed above, in the stf.inv file.

Reply:
Thanks,Emma. I submitted  the requested change